packages/fxa-settings/src/pages/Signin/index.tsx

```diff
@@ -0,0 +1,244 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useCallback, useState } from 'react';
+import { usePageViewEvent } from '../../lib/metrics';
+import { useFtlMsgResolver } from '../../models';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { RouteComponentProps, Link } from '@reach/router';
+import InputPassword from '../../components/InputPassword';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { ReactComponent as PocketLogo } from './pocket.svg';
+
+export type SigninProps = {
+  email: string;
+  isPasswordNeeded: boolean;
+  ServiceLogo?: React.ElementType;
+  serviceName?: string;
+};
+
+const Signin = ({
+  email,
+  isPasswordNeeded,
+  ServiceLogo,
+  serviceName = 'account settings',
+}: SigninProps & RouteComponentProps) => {
+  usePageViewEvent('signin', {
+    entrypoint_variation: 'react',
+  });
+
+  const isPocketClient = serviceName === 'Pocket';
+  const [error, setError] = useState('');
+  const [password, setPassword] = useState('');
+  const ftlMsgResolver = useFtlMsgResolver();
+  const localizedPasswordFormLabel = ftlMsgResolver.getMsg(
+    'password',
+    'Password'
+  );
+  const localizedPocketAriaLabel = ftlMsgResolver.getMsg(
+    'pocket-logo',
+    'Pocket logo'
+  );
+  const localizedLogoAriaLabel = ftlMsgResolver.getMsg(
+    'service-logo',
+    `${serviceName} logo`,
+    { serviceName }
+  );
+
+  const signInUsingLoggedInAccount = useCallback(() => {
+    // TODO: add in functionality to sign in using the logged in account
+    // return an error to be displayed if anythign goes wrong.
+  }, []);
+
+  const signInWithPassword = useCallback((email: string, password: string) => {
+    // TODO: add in the functionality to actually sign a user in using their password
+    // return an error to be displayed if anything goes wrong.
+  }, []);
+
+  // TODO: This page is also supposed to render the user card, complete with avatar.
+  // The avatar mixin is complex enough that I think it deserves its own ticket.
+
+  // TODO: The user-card mixin does some fancy footwork listening to see if you access token changes
+  // because it's possible for the access token to be invalidated when we display the avatar image.
+  // if we get the notification of a change, we re-render the card.
+
+  const onSubmit = useCallback(async () => {
+    try {
+      isPasswordNeeded
+        ? signInWithPassword(email, password)
+        : signInUsingLoggedInAccount();
+      // TODO: add message in Alert Bar re: success, then navigate to where appropriate
+    } catch (e) {
+      // TODO: metrics event for error
+      // TODO: Add in localized Alert Bar message for error
```

Comment:
I've created a `Banner` component I think we'll use instead of `AlertBar`. It'd be nice to use a shared message display between them but I think UX is going to want to see what that alert bar looks like across several pages before giving the thumbs up so I think we should plan to match parity as much as we can for now. 😩